### PR TITLE
update generator to maintain all plugin versions

### PR DESCRIPTION
**Pull request by lieut-data**
Update the generator to maintain all plugin versions (skipping pre-release, as usual). There are versions in here we will never actually release to the marketplace, but this helps the generator avoid having to download them again "to find out".

Fixes: https://mattermost.atlassian.net/browse/MM-21051

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,10 +40,10 @@ $ make test
 
 ### Updating plugins.json
 
-At the moment, the Marketplace simply points at the latest release of a fixed set of Mattermost plugins. In the future, this database will be fine-tuned to facilitate tracking multiple versions for the appropriate Mattermost server version. To update `plugins.json`, simply run:
+At the moment, the Marketplace simply points at the latest release of a fixed set of Mattermost plugins. In the future, this database will be fine-tuned to facilitate tracking multiple versions for the appropriate Mattermost server version. To update `plugins.json`, ensure you have [jq](https://stedolan.github.io/jq/) and [sponge](https://linux.die.net/man/1/sponge) installed and run:
 
 ```
-$ go run ./cmd/generator --github-token <your github token> --debug > plugins.json
+$ go run ./cmd/generator --github-token <your github token> --existing plugins.json --debug | jq | sponge plugins.json
```
Comment (hanzei) [inline README.md]: Maybe we add this as a `Makefile` target for now? I'm surely will come here every time to copy and past it.

File: cmd/generator/main.go [hunk 2]
```diff
@@ -177,45 +177,31 @@ func getReleasePlugins(ctx context.Context, client *github.Client, repositoryNam
...
-			}
+		pluginVersion, err := semver.Parse(plugin.Manifest.Version)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to parse release plugin version %s", plugin.Manifest.Version)
 		}
 
-		minServerVersionsSeen[releasePlugin.Manifest.MinServerVersion] = releasePlugin
-	}
+		if latestPlugin == nil || pluginVersion.GT(semver.MustParse(latestPlugin.Manifest.Version)) {
+			latestPlugin = plugin
```
Comment (marianunez) [inline cmd/generator/main.go]: What is `latestPlugin` used for?

Comment (lieut-data) [reply cmd/generator/main.go]: Good catch: this was from a version of my changes where I tracked the latest version of a plugin for handling changes of plugin ids from the same repository, but this become obsolete once we eliminated pre-release plugins, so I abandoned the extra complexity... but forgot to remove it here :)

Fixed!

**Comment by prapti** [comment]
Testing not required as discussed with Jesse. These changes are for internal tools and do not touch any application code. Removing QA Review label.

**Comment by hanzei** [comment]
@lieut-data To double check: I did not do a full review and just commented on a small thing that stand out to me. Do you want a full review from me?

**Comment by lieut-data** [comment]
Ah, my mistake @hanzei -- I had switched out @ali-farooq0 for you when I saw you comment. Will bring back @ali-farooq0 to get his feedback, especially since this builds on his changes.

**Comment by lieut-data** [comment]
As per @prapti's comment earlier, removing the QA Review label.

**Comment by lieut-data** [comment]
Waiting for Jira plugin update before merging this one.

**Pull request merged**